tsl/src/continuous_aggs/refresh.c

```diff
@@ -235,7 +319,7 @@ continuous_agg_refresh_with_window(const ContinuousAgg *cagg,
 		};
 
 		InternalTimeRange bucketed_refresh_window =
-			compute_bucketed_refresh_window(&invalidation, cagg->data.bucket_width);
+			compute_circumscribed_bucketed_refresh_window(&invalidation, cagg->data.bucket_width);
```

Comment:
When we process invalidations, since the refresh window was already aligned to buckets (line 467) and invalidation processing happens after that, can this ever happen:  bucketed_refresh_window != refresh_window?

Reply:
@gayyappan `refresh_window` can cover several invalidations, thus `bucketed_refresh_window`, which is calculated by `compute_circumscribed_bucketed_refresh_window`, might be much smaller than `refresh_window` (but it is never bigger).
For example
```
 [++++++++++++++++++++) - refresh_window
 |----|----|----|----|  - buckets
    [+++]    [+]        - invalidations
 [+++++++++)            - bucketed_refresh_window for the first invalidation
           [++++)       - bucketed_refresh_window for the second invalidation
```